src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -79,6 +87,25 @@ public User addUser(PrimaryKey id) {
 		return ResponseEntity.status(HttpStatus.OK).body(products);		
 	}
 	
+	public ResponseEntity<?> getCheckouts(int userId, int page, int nums) {
+		List<User> userCheckedOutMovies = userRepository.findCheckedOutMovies(userId, true, 
+				PageRequest.of(page, nums, Sort.by("numberOfFavorites").descending()));
+		HashMap<Integer, User> movieMap = new HashMap();
+		for(User u: userCheckedOutMovies) {
+			movieMap.put(u.getId().getProductId(), u);
+		}
+		//TODO: call search team's API here
```

Comment:
As far as I know search team's /search?keyworkd="abc" is working. Not sure though!